backend/src/libs/interfaces/service.interface.ts

```diff
@@ -1,5 +1,6 @@
 interface IService<T = unknown> {
   find(): Promise<T>;
+  findByEmail(email: string): Promise<T | null>;
```

Comment:
It should not be a part of all services - it should be implemented only upon user service
```suggestion
```